### PR TITLE
Use doas instead of sudo

### PR DESCRIPTION
**Pull request by nixbitcoin**
I have separated out these changes so we can merge them before https://github.com/fort-nix/nix-bitcoin/pull/307. I foresee a longer discussion and review for https://github.com/fort-nix/nix-bitcoin/pull/307 and I want to get these uncontroversial changes merged first.

### CODE REVIEW

**Review by erikarvstedt — APPROVED** [review]
ACK e287db453e454bf82a7c13ac3213ae8d6476425f

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
[Fixup](https://github.com/erikarvstedt/nix-bitcoin/commits/doas-ea).

File: README.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ NixOS modules
 
 Security
 ---
-* **Simplicity:** Only services you select in `configuration.nix` and their dependencies are installed, packages and dependencies are [pinned](pkgs/nixpkgs-pinned.nix), most packages are built from the [NixOS stable channel](https://github.com/NixOS/nixpkgs/tree/nixos-20.09), with a few exceptions that are built from the nixpkgs unstable channel, builds happen in a [sandboxed environment](https://nixos.org/manual/nix/stable/#conf-sandbox), code is continuously reviewed and refined.
+* **Simplicity:** Only services you select in `configuration.nix` and their dependencies are installed, packages and dependencies are [pinned](pkgs/nixpkgs-pinned.nix), [doas](https://github.com/Duncaen/OpenDoas) instead of [sudo](https://lobste.rs/s/efsvqu/heap_based_buffer_overflow_sudo_cve_2021#c_c6fcfa), most packages are built from the [NixOS stable channel](https://github.com/NixOS/nixpkgs/tree/nixos-20.09), with a few exceptions that are built from the nixpkgs unstable channel, builds happen in a [sandboxed environment](https://nixos.org/manual/nix/stable/#conf-sandbox), code is continuously reviewed and refined.
```
Comment (erikarvstedt) [inline README.md:83]:
```suggestion
* **Simplicity:** Only services you select in `configuration.nix` and their dependencies are installed, packages and dependencies are [pinned](pkgs/nixpkgs-pinned.nix), support for [doas](https://github.com/Duncaen/OpenDoas) ([sudo alternative](https://lobste.rs/s/efsvqu/heap_based_buffer_overflow_sudo_cve_2021#c_c6fcfa)), most packages are built from the [NixOS stable channel](https://github.com/NixOS/nixpkgs/tree/nixos-20.09), with a few exceptions that are built from the nixpkgs unstable channel, builds happen in a [sandboxed environment](https://nixos.org/manual/nix/stable/#conf-sandbox), code is continuously reviewed and refined.
```
Makes clear that sudo is still supported.

**Review by erikarvstedt — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (nixbitcoin) [reply README.md]: Fixed in b483362

Comment (erikarvstedt) [reply README.md]: "doas by default" is only valid for `secure-node`. I think the `support for doas` wording fits better in general.
You could add "doas by default" to the secure node description.

**Comment by erikarvstedt** [comment]
Needs rebase due to the `nix-bitcoin-services` -> `nbLib` renaming.

At a second glance, the prominent mention of `doas` in the secure node description looks weird, because it's a rather minor and non-bitcoin related feature. I'd remove it, but I'll ACK if you strongly prefer to keep it.

**Comment by erikarvstedt** [comment]
[Test failure fix](https://github.com/erikarvstedt/nix-bitcoin/commits/doas-ea2).

**Comment by nixbitcoin** [comment]
Ah, what a stupid mistake. In moments like these I'm thankful for our CI.

**Review by erikarvstedt — APPROVED** [review]
ACK 23f6fefd88c3ff1c8c0422c598a1bfc081785f7b

**Comment by erikarvstedt** [comment]
This PR has conflicts with #312 (`Refactorings, cleanups`).
Reviewing the rebase will be easier if we merge #312 first.

**Review by jonasnick — COMMENTED** [review]
Concept ACK, needs rebase

**Comment by nixbitcoin** [comment]
Rebased

**Review by jonasnick — COMMENTED** [review]
ACK 47d257ad3a68dd2f6b0e53f5088f9ef33f5bb8fa

**Review by jonasnick — APPROVED** [review]

**Pull request merged**